File: eclipse_plugin/com.sina.weibo.agent.plugin/src/com/sina/weibo/agent/plugin/Activator.java
package com.sina.weibo.agent.plugin;

import org.eclipse.ui.plugin.AbstractUIPlugin;
import org.osgi.framework.BundleContext;

/**
 * The activator class controls the plug-in life cycle
 */
public class Activator extends AbstractUIPlugin {

    // The plug-in ID
    public static final String PLUGIN_ID = "com.sina.weibo.agent.plugin";

    // The shared instance
    private static Activator plugin;

    /**
     * The constructor
     */
    public Activator() {
    }

    @Override
    public void start(BundleContext context) throws Exception {
        super.start(context);
        plugin = this;
        System.out.println("RunVSAgent Eclipse Plugin started");
        // TODO: Initialize extension host and VSCode extension integration
    }

    @Override
    public void stop(BundleContext context) throws Exception {
        plugin = null;
        super.stop(context);
        System.out.println("RunVSAgent Eclipse Plugin stopped");
        // TODO: Cleanup extension host and resources
    }

    /**
     * Returns the shared instance
     *
     * @return the shared instance
     */
    public static Activator getDefault() {
        return plugin;
    }
}

File: eclipse_plugin_wow/src.main.kotlin/com/sina/weibo/agent/eclipse/wow/plugin/Activator.kt
package com.sina.weibo.agent.eclipse.wow.plugin

import org.eclipse.ui.plugin.AbstractUIPlugin
import org.osgi.framework.BundleContext

class Activator : AbstractUIPlugin() {

    override fun start(context: BundleContext) {
        super.start(context)
        plugin = this
        println("RunVSAgent-wow plugin started")
    }

    override fun stop(context: BundleContext) {
        plugin = null
        super.stop(context)
    }

    companion object {
        @JvmStatic
        var plugin: Activator? = null
            private set
    }
}
